File: LeetCode/Java Solutions/Determine if Two Strings Are Close/determine.java
class Solution {
    public boolean halvesAreAlike(String s) {
        int mid = s.length() / 2, vowelCount1 = 0, vowelCount2 = 0;
        String vowels = "aeiouAEIOU";

        for (int i = 0; i < mid; i++) {
            if (vowels.indexOf(s.charAt(i)) >= 0) {
                vowelCount1++;
            }
            if (vowels.indexOf(s.charAt(i + mid)) >= 0) {
                vowelCount2++;
            }
        }

        return vowelCount1 == vowelCount2;
    }
}

File: LeetCode/Python Solutions/Determine if String Halves Are Alike/determine.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        def count_vowels(st):
            vowels = set("aeiouAEIOU")
            return sum(char in vowels for char in st)

        mid = len(s) // 2
        return count_vowels(s[:mid]) == count_vowels(s[mid:])
